server/src/main/java/keywhiz/commands/AddUserCommand.java

```diff
@@ -33,15 +35,25 @@ public AddUserCommand() {
 
   @Override protected void run(Bootstrap<KeywhizConfig> bootstrap, Namespace namespace,
                                KeywhizConfig config) throws Exception {
-    DataSource dataSource = config.getDataSourceFactory()
-      .build(new MetricRegistry(), "add-user-datasource");
 
     Console console = System.console();
     System.out.format("New username:");
     String user = console.readLine();
     System.out.format("password for '%s': ", user);
     char[] password = console.readPassword();
-    DSLContext dslContext = DSLContexts.databaseAgnostic(dataSource);
-    new UserDAO(dslContext).createUser(user, new String(password));
+    getUserDAO(bootstrap, config).createUser(user, new String(password));
+  }
+
+  @VisibleForTesting
+  static UserDAO getUserDAO(Bootstrap<KeywhizConfig> bootstrap, KeywhizConfig config) {
+    ManagedDataSource dataSource = config.getDataSourceFactory()
+        .build(new MetricRegistry(), "add-user-datasource");
+
+    Injector injector = InjectorFactory.createInjector(
+        config,
+        Environments.fromBootstrap(bootstrap),
+        dataSource);
+
+    return injector.getInstance(UserDAO.class);
   }
```

Comment:
Rather than assembling dependencies by hand, we now take advantage of the same Injector that the main service uses (data source bindings aside) via the InjectorFactory.